camviewer_ui_impl.py

```diff
@@ -17,6 +17,7 @@
 
 import sys
 import os
+import os.path
```

Comment:
I was under the mistaken impression that `import os` did not expose the utilities in `os.path`, because this is how all (most) other modules work. I'll remove the redundant import. I suspect this is a habit I picked up by reading code from other programmers.

There's some interesting implementation details about this I was unaware of, namely that the `os` module imports either `posixpath` or `ntpath` based on your OS and re-exports it as `os.path`, injecting it into `sys.modules` along the way. Pretty wild behavior, but I guess this is a reasonable way to aim for platform-independent libraries.

Reply:
The `os.path` platform specifics is a TIL for me as well

I'm pretty sure I'd have just tried to access the `path` as a submodule of `os`, but Zach is right that this is generally not suggested.  This access method only works if the parent module imports its submodules.  This isn't guaranteed to work for all module nor is it guaranteed to not change.